lib/reward.ts

```diff
@@ -0,0 +1,71 @@
+import type { Address, PublicClient } from 'viem'
+import { formatEther, encodeAbiParameters} from 'viem';
+import proxyAbi from '@/lib/proxy-abi';
+import { PROXY_CONTRACT_ADDRESS } from '@/lib/config';
+
+
+async function getRewarderAddress(
+  attestationType: string,
+  client: PublicClient,
+): Promise<Address> {
+  const [,rewarderAddress,] = await client.readContract({
+    address: PROXY_CONTRACT_ADDRESS,
+    abi: proxyAbi,
+    functionName: 'attestationTypes',
+    args: [attestationType],
+  }) as [any, Address, any];
+  return rewarderAddress;
+}
+
+
+const rewarderABI = [{
+  "type":"function",
+  "name":"rewardFor",
+  "inputs":[{
+    "name":"attestationData",
+    "type":"bytes",
+  }],
+  "outputs":[{
+    "type":"uint256",
+  }],
+  "stateMutability":"view"
+}]
```

Comment:
if you put `as const` after the ABI expression, viem will be able to infer types when you call `client.readContract`